flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/ResourceManagerGateway.java

```diff
@@ -71,6 +71,18 @@
 		@RpcTimeout Time timeout);
 
 	/**
+	 * Cancel the slot allocation requests from the resource manager.
+	 *
+	 * @param jobID JobID for which the JobManager was the leader
+	 * @param jobMasterId id of the JobMaster
+	 * @param allocationID The slot to request
+	 */
+	void cancelSlotRequest(
+		JobID jobID,
+		JobMasterId jobMasterId,
+		AllocationID allocationID);
```

Comment:
I think this could be `void cancelSlotRequest(AllocationID allocationId)`. Not sure why we need the other information.